bofire/data_models/objectives/desirabilities.py

```diff
@@ -0,0 +1,226 @@
+from typing import Literal, Optional, Union
+
+import numpy as np
+import pandas as pd
+import pydantic
+
+from bofire.data_models.objectives.identity import IdentityObjective
+
+
+class _SeriesNumpyCallable:
+    """Helper class to call numpy and torch functions with series or numpy arrays. matches __call__
+    signature of objectives."""
+
+    def __call__(
+        self, x: Union[pd.Series, np.ndarray], x_adapt
+    ) -> Union[pd.Series, np.ndarray]:
+        convert_to_series = False
+        if isinstance(x, pd.Series):
+            convert_to_series = True
+            name = x.name
+            x = x.values
+
+        y = self.call_numpy(x)
+
+        if convert_to_series:
+            return pd.Series(y, name=name)
+
+        return y
+
+    def call_numpy(self, x: np.ndarray) -> np.ndarray:
+        raise NotImplementedError()
+
+
+class DesirabilityObjective(IdentityObjective):
+    """Abstract class for desirability objectives. Works as Identity Objective"""
+
+    type: Literal["DesirabilityObjective"] = "DesirabilityObjective"  # type: ignore
+
```

Comment:
```suggestion
clip: bool = True
```

Reply:
Let us add the clip attribute diretly here in the abstract class. And let us then derive the `DesirabilityObjective` also from the `_SeriesNumpyCallable`.

Reply:
This should also resolve the lointing error below